pydantic/type_adapter.py

```diff
@@ -338,8 +338,7 @@ def json_schemas(
         key_map, definitions = schema_generator_instance.generate_definitions(inputs)
 
         json_schema: dict[str, Any] = {}
-        if definitions:
-            json_schema['$defs'] = definitions
+        json_schema['$defs'] = definitions
```

Comment:
Seems the `if definitions` condition is always `True`. So, I did this change to make the coverage 100%

Reply:
You need to readd it... 👀 
```
from pydantic.type_adapter import TypeAdapter

_, top_level_schema = TypeAdapter.json_schemas(
    [(int, 'validation', TypeAdapter(int))],
    ref_template='#/components/schemas/{model}',
)
print(top_level_schema)
# {}
```

Reply:
Yeah, you are right.
I've made a test from your example 😃